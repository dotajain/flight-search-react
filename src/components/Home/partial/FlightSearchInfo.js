import React from 'react';
import PropTypes from 'prop-types';

const FlightSearchInfo = ({ result }) => (
  <div className="col">
    <div className="flight">
      <small className="flight__number">{`${result.airline.code} - ${result
        .airline.number}`}</small>
      <div className="flight__airport-code">{`${result.origin
        .airportCode} > ${result.destination.airportCode}`}</div>

      <div className="flight__time">
        <span className="flight__time--label">Depart:</span>
        <span className="flight__time--value">{result.origin.time}</span>
      </div>

      <div className="flight__time">
        <span className="flight__time--label">Arrive:</span>
        <span className="flight__time--value">{result.destination.time}</span>
      </div>
    </div>
  </div>
);

FlightSearchInfo.propTypes = {
  result: PropTypes.object,
};

export default FlightSearchInfo;
