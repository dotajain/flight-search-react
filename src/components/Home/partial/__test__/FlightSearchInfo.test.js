import React from 'react';
import renderer from 'react-test-renderer';
import FlightSearchInfo from '../FlightSearchInfo';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly rendered FlightSearchInfo Component', () => {
  const component = renderer.create(<FlightSearchInfo result={result}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


const result = {
  'airline': {
    'name': 'Vistara',
    'code': '993',
    'number': 'UK',
  },
  'origin': {
    'airportCode': 'DEL',
    'terminal': '3',
    'time': '12:50',
  },
  'destination': {
    'airportCode': 'BOM',
    'time': '14:55',
  },
  'bookingclass': 'E',
  'depdate': '2017-09-15t1250',
  'arrdate': '2017-09-15t1455',
  'travelTime': '2h 5m',
  'fare': 2986,
};