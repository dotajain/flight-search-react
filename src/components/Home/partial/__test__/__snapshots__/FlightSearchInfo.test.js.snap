// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly rendered FlightSearchInfo Component 1`] = `
<div
  className="col"
>
  <div
    className="flight"
  >
    <small
      className="flight__number"
    >
      993 - UK
    </small>
    <div
      className="flight__airport-code"
    >
      DEL &gt; BOM
    </div>
    <div
      className="flight__time"
    >
      <span
        className="flight__time--label"
      >
        Depart:
      </span>
      <span
        className="flight__time--value"
      >
        12:50
      </span>
    </div>
    <div
      className="flight__time"
    >
      <span
        className="flight__time--label"
      >
        Arrive:
      </span>
      <span
        className="flight__time--value"
      >
        14:55
      </span>
    </div>
  </div>
</div>
`;
